packages/jest-worker/src/worker.js

```diff
@@ -91,8 +92,24 @@ export default class {
     // $FlowFixMe: wrong "ChildProcess.send" signature.
     child.send([CHILD_MESSAGE_INITIALIZE, false, this._options.workerPath]);
 
+    this._retries++;
     this._child = child;
     this._busy = false;
+
+    // If we exceeded the amount of retries, we will emulate an error reply
+    // coming from the child. This avoids code duplication related with cleaning
+    // the queue, and scheduling the next call.
+    if (this._retries > this._options.maxRetries) {
+      const error = new Error('Call retries were exceeded');
```

Comment:
Could this be a subclass of error with a `.type` field or something so that people can distinguish it from user errors?